test/acvp_test.c

```diff
@@ -71,7 +71,7 @@ static int pkey_get_bn_bytes(EVP_PKEY *pkey, const char *name,
     buf = OPENSSL_zalloc(sz);
     if (buf == NULL)
         goto err;
-    if (!BN_bn2binpad(bn, buf, sz))
+    if (BN_bn2binpad(bn, buf, sz) <= 0)
```

Comment:
Same problem here

Reply:
I changed to <= according to reutrn values check in /test/acvp_test.c:229, 231, 570, 572. There are a lot. Do we need to change them all?

Reply:
And /crypto/ec/ec_curve.c: 3404

Reply:
> Same problem here

Hmm, @kroeckx  but what the returned 0 indicates? A 0 byte sized number does not look like a number. Perhaps 0 cannot be returned at all? A zero BIGNUM should be encoded as a single zero byte, shouldn't it? So the return value should be 1.

Reply:
It's hard to see how a zero could be returned.

We should nonetheless try to be consistent in the return code handling.

Reply:
In https://github.com/openssl/openssl/pull/16943, the code invoking BN_bn2nativepad has a consistent handling way. As for  BN_bn2binpad, == -1, <0, and <=0 can be found in codebase.

Reply:
Agreed, they should be unified.  Not necessary for this PR IMO.